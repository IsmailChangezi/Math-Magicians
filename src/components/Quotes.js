import React, { useEffect, useState } from "react";

export default function Quotes() {
  const [quotes, setQuotes] = useState([]);

  useEffect(() => {
    const fetchData = () => {
      fetch("https://api.api-ninjas.com/v1/quotes?category=happiness", {
        headers: {
          "X-Api-Key": "6OswdSRYmeJgqtC+MLCqOA==iSGiiZldylP7Y018",
        },
      })
        .then((res) => res.json())
        .then((data) => setQuotes(data));
    };
    fetchData();
  }, []);
  console.log(quotes);
  return (
    <div>
      {quotes &&
        quotes.length > 0 &&
        quotes.map((items) => {
          return (
            <>
              <h1 key={items.author}>{items.quote}</h1>
              <h2>{items.author}</h2>
            </>
          );
        })}
    </div>
  );
}
