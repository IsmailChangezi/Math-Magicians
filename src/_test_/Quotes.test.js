import React from 'react';
import renderer from 'react-test-renderer';
import Quotes from '../components/Quotes';

it('renders when there is quote', () => {
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
