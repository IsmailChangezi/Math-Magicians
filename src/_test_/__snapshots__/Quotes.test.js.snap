// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is quote 1`] = `<div />`;
